tests/contrib/django_util/test_decorators.py

```diff
@@ -0,0 +1,243 @@
+        request = self.factory.get('/test')
+        request.session = mock.MagicMock()
+
+        credentials_mock = mock.Mock(
+            scopes=set(django.conf.settings.GOOGLE_OAUTH2_SCOPES))
+        credentials_mock.has_scopes.return_value = True
+        credentials_mock.invalid = False
+        OAuth2Credentials.from_json.return_value = credentials_mock
+
+        @decorators.oauth_enabled
+        def test_view(request):
+            return http.HttpResponse("test")
+
+        response = test_view(request)
+        self.assertEquals(response.status_code, http_client.OK)
+        self.assertEquals(response.content, b"test")
+        self.assertTrue(request.oauth.has_credentials())
+        self.assertIsNotNone(request.oauth.http)
+
+    @mock.patch('oauth2client.contrib.django_util.DictionaryStorage')
+    def test_specified_scopes(self, dictionary_storage_mock):
+        request = self.factory.get('/test')
+        request.session = mock.MagicMock()
+
+        credentials_mock = mock.Mock(
+            scopes=set(django.conf.settings.GOOGLE_OAUTH2_SCOPES))
+        credentials_mock.has_scopes = True
+        credentials_mock.is_valid = True
+        dictionary_storage_mock.get.return_value = credentials_mock
+
+        @decorators.oauth_enabled(scopes=['additional-scope'])
+        def test_view(request):
+            return http.HttpResponse("hello world")  # pragma: NO COVER
+
+        response = test_view(request)
+        self.assertEquals(response.status_code, http_client.OK)
+        self.assertIsNotNone(request.oauth)
+        self.assertFalse(request.oauth.has_credentials())
+
+
+class OAuth2RequiredDecoratorTest(TestWithDjangoEnvironment):
+
+    def setUp(self):
+        super(OAuth2RequiredDecoratorTest, self).setUp()
+        self.save_settings = copy.deepcopy(django.conf.settings)
+
+        if hasattr(django.conf.settings, 'GOOGLE_OAUTH2_STORAGE_MODEL'):
+            del django.conf.settings.GOOGLE_OAUTH2_STORAGE_MODEL
+
+        reload_module(oauth2client.contrib.django_util)
+        self.user = User.objects.create_user(
+            username='bill', email='bill@example.com', password='hunter2')
+
+    def tearDown(self):
+        super(OAuth2RequiredDecoratorTest, self).tearDown()
+        django.conf.settings = copy.deepcopy(self.save_settings)
+
+    def test_redirects_without_credentials(self):
+        request = self.factory.get('/test')
+        request.session = self.session
+
+        @decorators.oauth_required
+        def test_view(request):
+            return http.HttpResponse("test")  # pragma: NO COVER
+
+        response = test_view(request)
+        self.assertTrue(isinstance(response, http.HttpResponseRedirect))
+        self.assertEquals(parse.urlparse(response['Location']).path,
+                          "/oauth2/oauth2authorize/")
+        self.assertTrue(
+            "return_url=%2Ftest" in parse.urlparse(response['Location']).query)
+
+        self.assertEquals(response.status_code, 302)
```

Comment:
Symbolic constant rather than integer literal, please. I once heard and will never stop repeating "if you can't do arithmetic with them, they're not numbers".